equinix/fabric_port_read_schema.go

```diff
@@ -0,0 +1,350 @@
+			Computed:    true,
+			Description: "Port name",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port description",
+		},
+		"state": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port state",
+		},
+		"operation": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Port specific operational data",
+			Elem: &schema.Resource{
+				Schema: readPortOperationSch(),
+			},
+		},
+		"bandwidth": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Port bandwidth in Mbps",
+		},
+		"available_bandwidth": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Port available bandwidth in Mbps",
+		},
+		"used_bandwidth": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Port used bandwidth in Mbps",
+		},
+		"service_type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Port service type",
+		},
+		"account": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Customer account information that is associated with this port",
+			Elem: &schema.Resource{
+				Schema: readAccountSch(),
+			},
+		},
+		"change_log": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Captures port lifecycle change information",
+			Elem: &schema.Resource{
+				Schema: readChangeLogSch(),
+			},
+		},
+		"location": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Port location information",
+			Elem: &schema.Resource{
+				Schema: readLocationSch(),
+			},
+		},
+		"device": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Port device",
+			Elem: &schema.Resource{
+				Schema: readPortDeviceSch(),
+			},
+		},
+		"redundancy": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Port redundancy information",
+			Elem: &schema.Resource{
+				Schema: readPortsRedundancySch(),
+			},
+		},
+		"encapsulation": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Port encapsulation protocol",
+			Elem: &schema.Resource{
+				Schema: readFabricPortEncapsulation(),
+			},
+		},
+		"lag": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Port Lag",
+			Elem: &schema.Resource{
+				Schema: readFabricPortLag(),
+			},
+		},
+	}
+}
+
+func readGetPortsByNameQueryParamSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Query Parameter to Get Ports By Name",
+		},
+	}
+}
+
+func readFabricPortsResponseSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"data": {
+			Type:        schema.TypeList,
+			Computed:    true,
+			Description: "List of  Ports",
+			Elem: &schema.Resource{
+				Schema: readFabricPortResourceSchema(),
+			},
+		},
+		"local_var_optionals": {
```

Comment:
It seems `local_var_optionals` is not an API parameter and it was made up in the go sdk. It is not a very intuitive name for a field, also why do we need to group them? Wouldn't it be better to have a `Name` field directly without needing that schema Set ?